python/python/lance/torch/data.py

```diff
@@ -288,9 +292,12 @@ def __iter__(self):
             if self.rank is not None and self.world_size is not None:
                 rank = self.rank
                 world_size = self.world_size
-            else:
+            elif self._auto_detect_rank:
                 rank = get_global_rank()
                 world_size = get_global_world_size()
```

Comment:
could we just take `rank` and `world_size` in the args? if both are None, use inference, if they are set then use the provided values

Reply:
Or, use a `NO_AUTO_INIT = -1` value to avoid this branch?